### PR TITLE
fix: table footer background color

### LINKED ISSUE #2279: For table (with paginationType="server") footer, background color is missing.
For table (with paginationType="server") footer, background color is missing.

Actual - 
image

Expected - 
image

### PR DESCRIPTION
**Pull request by Sumitmaithani**
## Description
Fixes #2279 

## Component Checklist

- [ ] Update Component Status Page
- [ ] Perform Manual Testing in Other Browsers
- [ ] Add KitchenSink Story
- [ ] Add Interaction Tests (if applicable)
- [ ] Add changeset

### CODE REVIEW

**Comment by Sumitmaithani** [comment]
Hi @snitin315 @saurabhdaware can you please review this PR?

File: packages/blade/src/components/Table/types.ts [hunk 1]
```diff
@@ -354,6 +354,11 @@ type TablePaginationCommonProps = {
    * @default false
    */
   showLabel?: boolean;
+  /**
+   * The background color of the component.
+   * @default transparent
+   */
+  backgroundColor?: string;
```
Comment (saurabhdaware) [inline packages/blade/src/components/Table/types.ts:361]: Hi @Sumitmaithani , thanks for the PR.

So the issue is not that the backgroundColor prop is missing. Its expected that we set backgroundColor internally without taking it as a prop.

The issue is that the backgroundColor is not being set when the `paginationType="server"` prop is there.

You can replicate the issue by setting that prop on TablePagination component.

```jsx
<Table
          data={data}
          pagination={
            <TablePagination paginationType="server" showPageSizePicker showPageNumberSelector />
          }
        >
```

Comment (Sumitmaithani) [reply packages/blade/src/components/Table/types.ts:361]: Hi @saurabhdaware,

Thank you for helping me understand the issue. I am new to this codebase. From my understanding, I tried to reproduce the issue here: [https://stackblitz.com/edit/hhwvwi-wm3jcu?file=App.tsx](https://stackblitz.com/edit/hhwvwi-wm3jcu?file=App.tsx). It appears that the TablePagination component has a default transparent background color, regardless of whether we pass paginationType="server".

Could you clarify what it means when we set the background color internally without taking it as a prop? Does that mean we override its CSS? I apologize if I am misunderstanding or bothering you.

File: packages/blade/src/components/Table/TablePagination.web.tsx [hunk 2]
```diff
@@ -242,7 +243,7 @@ const _TablePagination = ({
       display="flex"
       flexDirection="row"
       padding={tablePagination.padding}
-      backgroundColor={tablePagination.backgroundColor}
+      backgroundColor={backgroundColor}
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Table/TablePagination.web.tsx:246]: Take the backgroundColor from `useTableContext` Similar to how we have done it on TableToolbar.web.tsx

Comment (Sumitmaithani) [reply packages/blade/src/components/Table/TablePagination.web.tsx:246]: Thanks @chaitanyadeorukhkar 🙌, Done.

File: packages/blade/src/components/Table/types.ts (hunk 1 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Table/types.ts]: So we by default want to set the backgroundColor. We dont want to accept that as prop from consumer.

You can check how its being set in TableHeader. I think right now we're setting the background color as transparent for footer but it should be same as header so I think we just need to change the background color

Comment (Sumitmaithani) [reply packages/blade/src/components/Table/types.ts]: Yes, I have fixed it as @chaitanyadeorukhkar mentioned. We are now taking the background color from `useTableContext`.

**Review by chaitanyadeorukhkar — APPROVED** [review]

File: packages/blade/src/components/Table/TablePagination.web.tsx [hunk 3]
```diff
@@ -242,7 +243,7 @@ const _TablePagination = ({
       display="flex"
       flexDirection="row"
       padding={tablePagination.padding}
-      backgroundColor={tablePagination.backgroundColor}
```
Comment (saurabhdaware) [inline packages/blade/src/components/Table/TablePagination.web.tsx:245]: Small change request: Can we also remove `backgroundColor` key from `tablePagination` object if its not being used anymore?

Comment (Sumitmaithani) [reply packages/blade/src/components/Table/TablePagination.web.tsx:245]: how I can miss this 🤦. Thanks @saurabhdaware  🙌, done.

**Review by saurabhdaware — APPROVED** [review]
Yay! Thanks for your contribution 🥳

**Pull request merged**